testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/management/cli/DeployAllDomainTestCase.java

```diff
@@ -123,69 +113,505 @@ public static void after() throws Exception {
+        ctx.handle("deployment disable --server-groups=" + sgTwo + ',' + sgOne + ' ' + cliTestAppEar.getName());
+
+        // Step 6) Check if all applications deployments is disabled in all server groups
+        infoUtils.checkDeploymentByInfo(sgOne, cliTestApp1War.getName(), ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestAnotherWar.getName(), ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestApp2War.getName(), NOT_ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestAppEar.getName(), ADDED);
+
+        infoUtils.checkDeploymentByInfo(sgTwo, cliTestApp1War.getName(), NOT_ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestAnotherWar.getName(), NOT_ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestApp2War.getName(), ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestAppEar.getName(), ADDED);
+
+        // Step 7) Enable all applications deployments for all server groups
+        ctx.handle("deployment enable-all --all-server-groups");
+
+        // Step 8) Verify if all applications deployments are enabled for all server groups
+        infoUtils.checkDeploymentByInfo(sgOne, cliTestApp1War.getName(), ENABLED);
+        infoUtils.checkExistInOutputMemory(cliTestAnotherWar.getName(), ENABLED);
+        infoUtils.checkExistInOutputMemory(cliTestApp2War.getName(), NOT_ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestAppEar.getName(), ENABLED);
 
+        infoUtils.checkDeploymentByInfo(sgTwo, cliTestApp1War.getName(), NOT_ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestAnotherWar.getName(), NOT_ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestApp2War.getName(), ENABLED);
+        infoUtils.checkExistInOutputMemory(cliTestAppEar.getName(), ENABLED);
+
+        // Step 9) Undeploy one application deployment
+        ctx.handle("deployment undeploy --server-groups=" + sgTwo + ' ' + cliTestApp2War.getName());
+        // Step 10) Check if selected application deployment is removed, but others still exist with right state
+        infoUtils.checkDeploymentByInfo(sgOne, cliTestApp1War.getName(), ENABLED);
+        infoUtils.checkExistInOutputMemory(cliTestAnotherWar.getName(), ENABLED);
+        infoUtils.checkMissingInOutputMemory(cliTestApp2War.getName());
+        infoUtils.checkExistInOutputMemory(cliTestAppEar.getName(), ENABLED);
+
+        infoUtils.checkDeploymentByInfo(sgTwo, cliTestApp1War.getName(), NOT_ADDED);
+        infoUtils.checkExistInOutputMemory(cliTestAnotherWar.getName(), NOT_ADDED);
+        infoUtils.checkMissingInOutputMemory(cliTestApp2War.getName());
+        infoUtils.checkExistInOutputMemory(cliTestAppEar.getName(), ENABLED);
+
+        // Step 11) Undeploy all applications deployments
+        ctx.handle("deployment undeploy * --all-relevant-server-groups");
+        // Step 12) Check if all applications deployments is gone
+        infoUtils.readDeploymentList();
```

Comment:
Here we could have:
List<String> lst = InfoUtils.readDeploymentList(cli);
InfoUtils.checkNotPresent(lst, cliTestApp1War.getName());